sync/src/chain.rs

```diff
@@ -437,6 +448,10 @@ impl ChainSync {
 		trace!(target: "sync", "{} -> NewBlock ({})", peer_id, h);
 		let header: BlockHeader = try!(header_rlp.as_val());
  		let mut unknown = false;
+		{
+			let peer = self.peers.get_mut(&peer_id).expect("ChainSync: unknown peer");
```

Comment:
i notice that there are others in the code, so happy for the whole lot to be cleared up in a later PR if preferable.